pkg/network/protocols/postgres/model_linux.go

```diff
@@ -0,0 +1,116 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux_bpf
+
+package postgres
+
+import (
+	"bytes"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/network/protocols"
+	"github.com/DataDog/datadog-agent/pkg/network/types"
+)
+
+type EventWrapper struct {
+	*EbpfEvent
+
+	operationSet bool
+	operation    Operation
+	tableNameSet bool
+	tableName    string
+}
+
+// ConnTuple returns the connection tuple for the transaction
+func (e *EventWrapper) ConnTuple() types.ConnectionKey {
+	return types.ConnectionKey{
+		SrcIPHigh: e.Tuple.Saddr_h,
+		SrcIPLow:  e.Tuple.Saddr_l,
+		DstIPHigh: e.Tuple.Daddr_h,
+		DstIPLow:  e.Tuple.Daddr_l,
+		SrcPort:   e.Tuple.Sport,
+		DstPort:   e.Tuple.Dport,
+	}
+}
+
+// getFragment returns the actual query fragment from the event.
+func (e *EbpfTx) getFragment() []byte {
+	if e.Original_query_size == 0 {
+		return nil
+	}
+	if e.Original_query_size > uint32(len(e.Request_fragment)) {
+		return e.Request_fragment[:len(e.Request_fragment)]
+	}
+	return e.Request_fragment[:e.Original_query_size]
+}
+
+// Operation returns the operation of the query (SELECT, INSERT, UPDATE, DROP, etc.)
```

Comment:
what about `ALTER` statement, are this not meant to be tracked?

Reply:
https://github.com/DataDog/datadog-agent/pull/25416#discussion_r1592535282